### PR TITLE
GFORMS-383 - Introduce IncludeIf functionality at field level

### PR DESCRIPTION
**Pull request by josef-vlach**
Validation fixes for revealing choice and group components

### CODE REVIEW

File: app/uk/gov/hmrc/gform/models/VisibilityFilter.scala [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    group.copy(
+      fields = group.fields.filter(isVisible)
+    )
+
+  def stripHiddenFormComponents(formModel: FormModel[P]): FormModel[P] =
+    formModel.map[P] { singleton =>
+      val visibleFields: List[FormComponent] =
+        singleton.page.fields
+          .map {
+            case fc @ IsRevealingChoice(rc) => fc.copy(`type` = stripHiddenFormComponentsFromRevealingCoice(rc))
```
Comment (girishkamat) [inline app/uk/gov/hmrc/gform/models/VisibilityFilter.scala:49]: Suggestion: for some of the deeply nested copies, monocle can simplify the code i guess.

File: app/uk/gov/hmrc/gform/models/VisibilityFilter.scala [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.gform.models
+
+import uk.gov.hmrc.gform.models.optics.{ DataOrigin, FormModelVisibilityOptics }
+import uk.gov.hmrc.gform.sharedmodel.formtemplate.{ FormComponent, FormPhase, Group, IsGroup, IsRevealingChoice, RevealingChoice }
+
+class VisibilityFilter[D <: DataOrigin, P <: PageMode](
```
Comment (girishkamat) [inline app/uk/gov/hmrc/gform/models/VisibilityFilter.scala:22]: FormComponentVisibilityFilter?

Comment (josef-vlach) [reply app/uk/gov/hmrc/gform/models/VisibilityFilter.scala:22]: What about 
```
app/uk/gov/hmrc/gform/models/formcomponent/VisibilityFilter.scala
```
instead?

Comment (josef-vlach) [reply app/uk/gov/hmrc/gform/models/VisibilityFilter.scala]: Nevermind, `FormComponentVisibilityFilter` is it.

File: app/uk/gov/hmrc/gform/models/VisibilityFilter.scala (hunk 1 above)
Comment (josef-vlach) [reply app/uk/gov/hmrc/gform/models/VisibilityFilter.scala]: ok, quicklens is it than (since we already use it elsewhere)

**Review by girishkamat — APPROVED** [review]

**Pull request merged**